### DISCUSSION TITLE
maybe these components are not compatible after all

### DISCUSSION THREAD

**Discussion by JungleJim440** [Bodmer/TFT_eSPI#3045] [Compatible displays and setup files]
Rather than bore you all with the many rabbit holes I have been down trying to make this work I shall simply list the major hardware components and perhaps I can get some feedback from the more experienced than I on whether or not I am doomed to fail.  [ others have surely been here before me ]

I am using an ST7796S based 3.5" TFT display upon which I can run simple graphics tests with Adruino_ST7796S_kbv.h
I would like to use some sprites on this 320x480 TFT display so ultimately I am thinking about TFT_eSPI.h
and...as this display supports capacitive touch,  that too be a road I shall crawl down at some point.

I am using a RPi Pico  (for me a big set up from an Arduino UNO or NANO)

I am programming with the Arduino IDE
my code validates, and based up some serial print messages along the way my void loop() is marching along as expected.

Thus far I have NOT been able to execute so much as a successful "tft.fillScreen(TFT_RED);" when using TFT_eSPI.h
I have been just fine with a "tft.fillScreen(ST7796S_GREEN);" when using Adafruit_ST7796S_kbv.h

thanks ahead of time to any and all who politely choose to help me pull my head of of my arse :)

the attached text file is for the "myCustom_Setup" file created in compliance with Bodmer "TIPs" section as a guide,
and which is being call upon from within TFT_eSPI "User_Setup_Select" as shown here
" #include <../TFT_eSPI_Setups/JimiG_RPiPico_ST7796_custom_setup440.h> "

[JimiG_RPiPico_ST7796_custom_setup440.txt](https://github.com/Bodmer/TFT_eSPI/files/13693959/JimiG_RPiPico_ST7796_custom_setup440.txt)

**Comment by Bodmer** [comment]
GPIO 19 is MOSI
GPIO 16 is MISO

You have these swapped at the moment. The other alloations look good. BTW, you need to use Earle Philhower's RP2040 board package.

**Reply by JungleJim440** [reply]
Thank you for the unexpectedly rapid response.
I do indeed have Earle Philhower's RP2040 board package(v3.6.2) loaded up within my Arduino IDE board Manager.(v2.2.1)
The pin arrangement as-is works with Adafruit_ST7796S_kbv.h,  but I went ahead and tried swapping in the custom setup file for TFT_eSPI anyway.   no change.   hmmm....

I will be away for a couple of days,  off to meet my new grand-daughter.
I shall re-visit this little project when I get back.

**Reply by Bodmer** [reply]
Adafruit_ST7796S_kbv may bit-bash the SPI signals so that is why it works. The hardware SPI however needs certain pins to be used and GPIO cannot be used for MOSI. [See here](https://datasheets.raspberrypi.com/pico/pico-datasheet.pdf) for default pin allocations (MOSI - SPI0 TX).

You can use any pins if you invoke the PIO implemented SPI port. Add following line to the setup file:
#define RP2040_PIO_SPI

Run the Read_User_Setup sketch and look at the serial output to see if the correct settings are being picked up by the compiler. A common mistake is to forget to [coment out this line](https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup_Select.h#L27).

**Comment by JungleJim440** [comment]
Sorry for the several days delay in getting you this response.  I returned home to several days without electricity/phone/internet caused by heavy storm damage in the area.   Thank you sir,  "#define RP2040_PIO_SPI"  did the trick.    I am off to the races !
